extensions/panache/mongodb-panache-common/runtime/src/main/java/io/quarkus/mongodb/panache/common/reactive/runtime/ReactiveMongoOperations.java

```diff
@@ -216,8 +217,20 @@ public Uni<Void> delete(Object entity) {
     public ReactiveMongoCollection mongoCollection(Class<?> entityClass) {
         MongoEntity mongoEntity = entityClass.getAnnotation(MongoEntity.class);
         ReactiveMongoDatabase database = mongoDatabase(mongoEntity);
-        if (mongoEntity != null && !mongoEntity.collection().isEmpty()) {
-            return database.getCollection(mongoEntity.collection(), entityClass);
+        if (mongoEntity != null) {
+            ReactiveMongoCollection collection = mongoEntity.collection().isEmpty()
+                    ? database.getCollection(entityClass.getSimpleName(), entityClass)
+                    : database.getCollection(mongoEntity.collection(), entityClass);
+            if (!mongoEntity.readPreference().isEmpty()) {
+                try {
+                    collection = collection.withReadPreference(ReadPreference.valueOf(mongoEntity.readPreference()));
+                } catch (IllegalArgumentException iae) {
+                    throw new IllegalArgumentException("Illegal read preference " + mongoEntity.readPreference()
+                            + " configured in the @MongoEntity annotation of " + entityClass.getName() + "."
+                            + " Supported values are primary|primaryPreferred|secondary|secondaryPreferred|nearest");
```

Comment:
Just asking: you cannot extract the values from the enum? (I haven't checked)

Reply:
No, this is not en enum but a regular class.